nemo_skills/mcp/servers/arxiv_tool.py

```
@@ -0,0 +1,198 @@
+            )
+        return results
+
+    try:
+        results = _with_retry(_fetch)
+        if not results:
+            return "No papers found for this query."
+        result_str = "\n---\n".join(results)
+        if len(_cache) < _CACHE_MAX_SIZE:
+            _cache[key] = result_str
+        return result_str
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)